acl/src/main/java/org/springframework/security/acls/jdbc/BasicLookupStrategy.java

```diff
@@ -152,8 +137,9 @@ public BasicLookupStrategy(DataSource dataSource, AclCache aclCache,
 	 * @param aclAuthorizationStrategy authorization strategy (required)
 	 * @param grantingStrategy the PermissionGrantingStrategy
 	 */
-	public BasicLookupStrategy(DataSource dataSource, AclCache aclCache,
-			AclAuthorizationStrategy aclAuthorizationStrategy, PermissionGrantingStrategy grantingStrategy) {
+	public  BasicLookupStrategy(DataSource dataSource, AclCache aclCache,
```

Comment:
Adding a constructor parameter breaks backward compatibility, so I would recommend that this be moved to a setter. 

Additionally, it's quite common in Spring Security for required members to be constructor parameters and optional members to have setters.

Reply:
This setter should also confirm that the provided strategy is not null.